### PR TITLE
Fix release command with private key case

### PR DESCRIPTION
**Pull request by alexandergoncharov**
**Issue**: We lose private key during release for RN from custom path because there should be root folder "CodePush" for React-Native app. (For Cordova it should be "www").  

**Suggestion fix**: In private key enabled case we should check that it is React-Native app and the last folder isn't "CodePush". In this case we should copy all contents to temp folder with root "CodePush" folder and then make release from here.

### CODE REVIEW

**Comment by owenniblock** [comment]
Thanks for contributing @alexandergoncharov - we'll take a look at this as soon as we can.

Would it be possible to add some tests for this functionality?

I don't know the `codepush` command well enough to say if what we're doing here is good or not so I'll seek advise but some initial questions you might be able to answer:
- Is it possible that this change will break the command for existing react-native users? (Would it make sense for this functionality to be on a switch or would it be better if it _does_ break and users have to fix it)
- You mention that for Cordova it should be `www`? Is this already handled by the cli? Or should we add handling for that as part of this PR?

Thanks!

**Comment by alexandergoncharov** [comment]
Hi @owenniblock ,

Yeah, I think that I can add tests for this a bit later. Sorry for delay.

About your questions:
1) No, it shouldn't break any commands/logic for users. This changes just fix path for releasing context to make compatibility with code-push SDK for RN which require for working sign feature. Forgot to mention that this fix is for https://github.com/Microsoft/appcenter-cli/issues/456

2) Yes, that's correct. No, cli doesn't handle this because Cordova automatically build content in `www` folder. And users use this folder for releasing.

Also, I have some improvements for code organisation for my PR. I'll update it today.

Thanks,
Alexander

**Comment by owenniblock** [comment]
Hi @alexandergoncharov - is this ready for a re-review? Or are you still working on it? :)

**Comment by alexandergoncharov** [comment]
Hi @owenniblock ,

Code is ready for review. For now I'm working on tests for this.

**Comment by alexandergoncharov** [comment]
Hi @owenniblock ,

Sorry for delay. I added test for implemented logic. I think it is ready for review. :)

Please let me know if you have any questions.

**Comment by owenniblock** [comment]
Thanks @alexandergoncharov - sorry I've been away. I'll take a look at this as soon as possible.

File: src/commands/codepush/lib/file-utils.ts [hunk 1]
```diff
@@ -27,6 +28,30 @@ export function copyFileToTmpDir(filePath: string): string {
...
+    fs.mkdirSync(tmpUpdateContentsPath);
+
+    if (isDirectory(updateContentsPath)) {
+      await pfs.cp(normalizePath(updateContentsPath), normalizePath(tmpUpdateContentsPath));
+    } else {
+      const targetFileName = path.parse(updateContentsPath).base;
+      await pfs.cpFile(normalizePath(updateContentsPath), path.join(tmpUpdateContentsPath, targetFileName));
+    }
+
+    return Promise.resolve(tmpUpdateContentsPath);
```
Comment (max-mironov) [inline src/commands/codepush/lib/file-utils.ts]: nit: no need to return `Promise` from the `async` function, should be wrapped automatically in a promise.

Comment (alexandergoncharov) [reply src/commands/codepush/lib/file-utils.ts]: @max-mironov Yeah, That's right. Fixed. Thanks!

**Pull request merged**